main.go

```diff
@@ -0,0 +1,393 @@
+			glog.Fatal(err)
+		}
+		orgName := strings.TrimPrefix(response.Name, "//cloudresourcemanager.googleapis.com/organizations/")
+		glog.V(20).Infof("     Found Organization %s", orgName)
+		allOrganizations[fmt.Sprintf("organizations/%s", orgName)] = response
+	}
+
+	// use asset-ap API to recall/find the projects under the provided organizationID.
+	//  we do not use the resource manager api here since that would return all the projects
+	//   the caller has access to, not just restricted to the organization we want.
+	//    https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#ProjectsListCall
+	// emptyQuery: search all projects https://cloud.google.com/asset-inventory/docs/searching-resources#how_to_construct_a_query
+	emptyQuery := ""
+	assetType = "cloudresourcemanager.googleapis.com/Project"
+	req = &assetpb.SearchAllResourcesRequest{
+		Scope:      fmt.Sprintf("organizations/%s", *organization),
+		Query:      emptyQuery,
+		AssetTypes: []string{assetType},
+	}
+
+	it = assetClient.SearchAllResources(ctx, req)
+
+	for {
+		response, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			glog.Fatal(err)
+		}
+		projectID := strings.TrimPrefix(response.Name, "//cloudresourcemanager.googleapis.com/projects/")
+		glog.V(20).Infof("     Found projectID %s", projectID)
+		allProjects[projectID] = response
+	}
+
+	glog.V(20).Infoln("      Getting Users")
+
+	// initialize the Workspace admin client first
+	//  this client will be used to find users in a domain
+	var adminService *admin.Service
+	if *impersonatedServiceAccount != "" {
+		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: *impersonatedServiceAccount,
+			Scopes:          []string{admin.AdminDirectoryUserReadonlyScope},
+			Subject:         *subject,
+		})
+		if err != nil {
+			glog.Fatal(err)
+		}
+		adminService, err = admin.NewService(ctx, option.WithTokenSource(ts))
+		if err != nil {
+			glog.Fatal(err)
+		}
+	} else {
+		cred, err := google.CredentialsFromJSONWithParams(ctx, svcAccountJSONBytes, google.CredentialsParams{
+			Scopes:  []string{admin.AdminDirectoryUserReadonlyScope},
+			Subject: *subject,
+		})
+		if err != nil {
+			glog.Fatal(err)
+		}
+		adminService, err = admin.NewService(ctx, option.WithCredentials(cred))
+		if err != nil {
+			glog.Fatal(err)
+		}
+	}
+	// if you want to narrow the search to a subset of users, apply a filter
+	//  https://developers.google.com/admin-sdk/directory/v1/guides/search-users
+	//	searchFilter := "isAdmin=false"
+
+	// for now, look for all users
+	searchFilter := ""
+
+	pageToken := ""
+	q := adminService.Users.List().Customer(*cx).Query(searchFilter)
+	for {
+		if pageToken != "" {
+			q = q.PageToken(pageToken)
+		}
+		r, err := q.Do()
+		if err != nil {
+			glog.Fatal(err)
+		}
+		for _, u := range r.Users {
+			glog.V(20).Infof("      Found User: %s", u.PrimaryEmail)
+			// just use goroutines to print out the visibility
```

Comment:
Clarify this comment, it should be a proper statement with capitalization and punctuation.